services/basic_types.proto

```diff
@@ -90,422 +139,616 @@ enum BlockHashAlgorithm {
 
     /**
-     * If true then the transfer is expected to be an approved allowance and the
-     * accountID is expected to be the owner. The default is false (omitted).
+     * An approved allowance flag.<br/>
+     * If true then the transfer is expected to be an approved allowance.
+     * <p>
+     * If set, `accountID` SHALL be the owner that previously approved
+     * the allowance.<br/>
+     * The default value SHALL be false (unset).
      */
     bool is_approval = 3;
 }
 
 /**
- * A list of accounts and amounts to transfer out of each account (negative) or into it (positive).
+ * A list of accounts and amounts to transfer.
+ *
+ * Each `AccountAmount` SHALL specify the account and the amount to
+ * send(negative) or receive(positive).<br/>
+ * Each `TransferList` SHALL be contained in another message that contains
+ * other details required to complete a transfer. This is typically a
+ * `CryptoTransferTransactionBody` or `TransactionRecord`.<br/>
+ * The `TransferList` SHALL only be used for HBAR transfers. Other token types
+ * MUST use the `TokenTransferList` message.
  */
 message TransferList {
     /**
-     * Multiple list of AccountAmount pairs, each of which has an account and
-     * an amount to transfer into it (positive) or out of it (negative)
+     * A list of AccountAmount pairs.<br/>
+     * Each entry in this list is an account and an amount to transfer
+     * into it (positive) or out of it (negative)
      */
     repeated AccountAmount accountAmounts = 1;
 }
 
 /**
- * A sender account, a receiver account, and the serial number of an NFT of a Token with
- * NON_FUNGIBLE_UNIQUE type. When minting NFTs the sender will be the default AccountID instance
- * (0.0.0) and when burning NFTs, the receiver will be the default AccountID instance.
+ * A NFT transfer.<br/>
+ * This refers to a sender account, a receiver account, and the serial number
+ * of an NFT to transfer from sender to receiver.
+ *
+ * Each `NftTransfer` SHALL be contained in another message (typically
+ * `TokenTransferList`) that details which `Token` type applies to this NFT
+ * transfer.
  */
 message NftTransfer {
     /**
-     * The accountID of the sender
+     * An Account identifier for the sender.
      */
     AccountID senderAccountID = 1;
 
     /**
-     * The accountID of the receiver
+     * An Account identifier for the receiver.
      */
     AccountID receiverAccountID = 2;
 
     /**
-     * The serial number of the NFT
+     * A serial number for the NFT to transfer.
      */
     int64 serialNumber = 3;
 
     /**
-     * If true then the transfer is expected to be an approved allowance and the
-     * senderAccountID is expected to be the owner. The default is false (omitted).
+     * An approved allowance flag.<br/>
+     * If true then the transfer is expected to be an approved allowance.
+     * <p>
+     * If set, `senderAccountID` SHALL be the owner that previously approved
+     * the allowance.<br/>
```

Comment:
Is it worth specifying the spender account must have paid the transaction ?

Reply:
If it's a requirement for this transaction, then yes.
is this correct?
`If set, the `senderAccountID` MUST be the "spender" account for the transaction`

Reply:
It is the "payer" should be the "spender" who is already given allowance by the owner in previous transaction.

Reply:
@Neeharika-Sompalli Any thoughts on this?